core/rawdb/accessors_chain.go

```diff
@@ -607,6 +607,37 @@ func ReadReceipts(db ethdb.Reader, hash common.Hash, number uint64, config *para
 	return receipts
 }
 
+// ReadLogs retrieves the logs for all transactions in a block. The log fields
+// are populated with metadata. In case the receipts or the block body
+// are not found, a nil is returned.
+func ReadLogs(db ethdb.Reader, hash common.Hash, number uint64) [][]*types.Log {
```

Comment:
Triage discussion, something like 
```suggestion
type LogFilterFn func(receipts []*types.ReceiptLogs) bool

func ReadLogs(db ethdb.Reader, hash common.Hash, number uint64, filterFn LogFilterFn) [][]*types.Log {
```
If the filter function returns `false`, we can omit the ReadBody and deriving the fields.